### PR TITLE
[CIS-371] LLC for macOS

### PR DESCRIPTION
**Pull request by dmigach**
We took this stale PR https://github.com/GetStream/stream-chat-swift/pull/811, rebased it on main, and fixed several additional things to make it work on macOS.
One of them is that we changed where local DB is stored for macOS because documents directory is a really bad place for it

### CODE REVIEW

File: StreamChat.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6878,18 +6922,28 @@
...
 			buildSettings = {
+				ARCHS = (
+					arm64,
+					armv7,
+					armv7s,
+					x86_64,
+				);
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = D48F367GRM;
+				DEVELOPMENT_TEAM = EHV7XZLAHA;
```
Comment (VojtaStavik) [inline StreamChat.xcodeproj/project.pbxproj:6932]: This is probably committed by mistake?

File: Sources/StreamChat/APIClient/APIClient_Tests.swift [hunk 2]
```diff
@@ -333,16 +333,19 @@ extension URLSessionConfiguration {
...
-//            && connectionProxyDictionary == otherConfiguration.connectionProxyDictionary
             && waitsForConnectivity == otherConfiguration.waitsForConnectivity
             && isDiscretionary == otherConfiguration.isDiscretionary
             && sharedContainerIdentifier == otherConfiguration.sharedContainerIdentifier
+        
+        #if os(iOS)
+        return commonEquatability
+            && multipathServiceType == otherConfiguration.multipathServiceType
             && sessionSendsLaunchEvents == otherConfiguration.sessionSendsLaunchEvents
+        #endif
```
Comment (VojtaStavik) [inline Sources/StreamChat/APIClient/APIClient_Tests.swift:347]: nice solution 👍

File: .github/workflows/macos-checks.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+name: MacOS checks
```
Comment (VojtaStavik) [inline .github/workflows/macos-checks.yml:1]: `macOS` 😱

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### ✅ Added
 - Add `areTypingEventsEnabled`, `areReactionsEnabled`, `areRepliesEnabled`, `areReadEventsEnabled`, `areUploadsEnabled` to `ChatChannelListController` [#1085](https://github.com/GetStream/stream-chat-swift/pull/1085)
+- Add possibility to build the framework on macOS
+    [#1132](https://github.com/GetStream/stream-chat-swift/pull/1132)
```
Comment (VojtaStavik) [inline CHANGELOG.md:11]: -> "to build the `StreamChat` framework on macOS"

**Review by VojtaStavik — APPROVED** [review]
Looks great, thank you 👍

File: StreamChat.xcodeproj/project.pbxproj (hunk 1 above)
Comment (dmigach) [reply StreamChat.xcodeproj/project.pbxproj:6932]: Yeah leftovers from the initial PR

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Add `areTypingEventsEnabled`, `areReactionsEnabled`, `areRepliesEnabled`, `areReadEventsEnabled`, `areUploadsEnabled` to `ChatChannelListController` [#1085](https://github.com/GetStream/stream-chat-swift/pull/1085)
 - Add `ImageCDN` protocol to improve work with image cache and thumbnails [#1111](https://github.com/GetStream/stream-chat-swift/pull/1111)
 - Add missing APIs `open` of `ComposerVC`. Including the delegate implementations and showing the suggestions as a child view controller. [#1140](https://github.com/GetStream/stream-chat-swift/pull/1140)
+- Add possibility to build the StreamChat framework on macOS
+    [#1132](https://github.com/GetStream/stream-chat-swift/pull/1132)
```
Comment (VojtaStavik) [inline CHANGELOG.md:13]: StreamChat -> `StreamChat` ;-)

**Pull request merged**